src/com/sun/jna/Function.java

```diff
@@ -274,12 +285,15 @@ public Object invoke(Class returnType, Object[] inArgs) {
     public Object invoke(Class returnType, Object[] inArgs, Map options) {
         // Clone the argument array to obtain a scratch space for modified
         // types/values
-        Object[] args = { };
+        if (argTable.get() == null) {
+          createArgTable();
+        }
+        Object[] args = argTable.get()[0];
```

Comment:
Ya, but then there are a lot more +1 and -1 using the table.  Less confusing this way.